packages/fxa-auth-server/lib/routes/password.ts

```diff
@@ -852,74 +848,69 @@ module.exports = function (
           }),
         },
       },
-      handler: async function (request) {
+      handler: async function (request: AuthRequest) {
         log.begin('Password.forgotVerify', request);
-        const passwordForgotToken = request.auth.credentials;
-        const code = request.payload.code;
-        const accountResetWithRecoveryKey =
-          request.payload.accountResetWithRecoveryKey;
+        const passwordForgotToken = request.auth.credentials as any;
+        const { code, accountResetWithRecoveryKey } = request.payload as {
+          code: string;
+          accountResetWithRecoveryKey?: boolean;
+        };
 
         const { deviceId, flowId, flowBeginTime } = await request.app
           .metricsContext;
 
-        let accountResetToken;
-
-        return Promise.all([
+        await Promise.all([
           request.emitMetricsEvent('password.forgot.verify_code.start'),
           customs.check(
             request,
             passwordForgotToken.email,
             'passwordForgotVerifyCode'
           ),
-        ])
-          .then(() => {
-            if (
-              butil.buffersAreEqual(passwordForgotToken.passCode, code) &&
-              passwordForgotToken.ttl() > 0
-            ) {
-              return db.forgotPasswordVerified(passwordForgotToken);
-            }
+        ]);
 
-            return failVerifyAttempt(passwordForgotToken).then(() => {
-              throw error.invalidVerificationCode({
-                tries: passwordForgotToken.tries,
-                ttl: passwordForgotToken.ttl(),
-              });
-            });
-          })
-          .then((result) => {
-            accountResetToken = result;
-            return Promise.all([
-              request.propagateMetricsContext(
-                passwordForgotToken,
-                accountResetToken
-              ),
-              db.accountEmails(passwordForgotToken.uid),
-            ]);
-          })
-          .then(([_, emails]) => {
-            if (accountResetWithRecoveryKey) {
-              // To prevent multiple password change emails being sent to a user,
-              // we check for a flag to see if this is a reset using an account recovery key.
-              // If it is, then the notification email will be sent in `/account/reset`
-              return Promise.resolve();
-            }
+        let accountResetToken;
+        if (
```

Comment:
Inverting the logic for the if statements seems a bit cleaner (ie put raising the error first), and would let us use a const for acccount reset token. Since this is refactor, I can see not wanting to change this though...